### PR TITLE
STRM-1269: Add manifest to enable CI

### PR DESCRIPTION
**Pull request by rakeshcusat**
Why?
 - We needed to add the manifest file so CI trigger
   the build when we create a release tag in github

**Please** add a meaningful description for your change here

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by rakeshcusat** [comment]
meow @tweise

**Comment by tweise** [comment]
OK, so we would really need this script in the `release-2.7.0-lyft` branch as that is what we will want to build. We will never do a build from master - is the script required in master for the CI to function?

Then we should not have a requirements.txt file because the dependencies are already defined in setup.py and they can change at any time when upstream changes, which will leave us with an outdated requirements.txt.

Can we instead not just execute the Beam build?

**Comment by tweise** [comment]
The project build file is here, it show the commands that are run to make the standard Beam Python SDK build: https://github.com/lyft/beam/blob/release-2.7.0-lyft/sdks/python/build.gradle#L52

**Comment by rakeshcusat** [comment]
@tweise We can select the right branch when we are releasing the code. Github does provide that option. We would just select the `release-2.7.0-lyft` branch. I think CI should be fine to pick that branch.
I think CI also runs the test cases and hence it needs the requirements.txt to setup the environment before running the test. I can double check with CI people if there is any workaround.
I don't think CI will pick the beam build for Python project. We need to talk to CI folks before we can make any changes.

I just check the Gradle file and it doesn't run the test before creating the distribution package. We configure manifest.yaml to not run any test case during the release process. But do we really want to avoid running test cases?

**Comment by tweise** [comment]
I don't think we need to run the tests in the fork, or at least we need to be able to skip them to not block ourselves at the time of creating the release.

File: sdks/python/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: apache_beam
+description: Lyft version of Apache Beam python package
+repository: git@github.com:lyft/beam.git
+credentials: true
+builder:
+  name: s2i
+  params:
+    builder: lyft/s2ipythonlibrary:494bddd033e63c155aababf999650fea120bb664
+tests:
+  - name: unit
```
Comment (tweise) [inline sdks/python/manifest.yaml]: isn't this what triggers the tests

Comment (tweise) [reply sdks/python/manifest.yaml]: See here for an example how the test command is explicit (though it is a Java project): https://github.com/lyft/streamingplatform/blob/master/manifest.yaml#L23

**Comment by tweise** [comment]
@rakeshcusat picking the branch to create the release from the HEAD SHA should work fine, but if there is no manifest in that SHA (because it is in master instead), won't the build fail?

**Comment by tweise** [comment]
Since it may inform some of the choices, I also wanted to mention what the ideal outcome (not necessarily now) would be. Beam has multiple languages (Java, Python, Go), each with their respective tooling, and Gradle is used as the build tool that pulls it all together. A Beam build is normally run from the top level directory and specific components can be built by providing the respective Gradle command line options. For Java, you can see that here: https://github.com/lyft/streamingplatform/blob/master/scripts/beam-release.sh#L48

Ideally we would like to be able to build all the pieces from the CI, if the infrastructure would allow to setup a top level manifest that way. I understand that may not be possible currently, and probably there are multiple reasons such as that we don't have a base container with the tooling for multiple languages etc. But perhaps it would be good to discuss this with someone from the CI team to make sure we are not missing anything.

**Comment by tweise** [comment]
@asottile what is your take on setting this up so that the CI builder just calls a script that encapsulates how the project is built (since that isn't something we control for a fork)?

**Comment by tweise** [comment]
Since we are working from the lyft branch, I believe the manifest needs to live there so that it is part of the commit that we are going to release. I'm not sure we even need this in master.

**Comment by rakeshcusat** [comment]
@tweise We don't need this on the master branch if we are not releasing it from the master branch.

**Review by tweise — APPROVED** [review]

**Comment by rakeshcusat** [comment]
🔨

**Comment by rakeshcusat** [comment]
💨

**Pull request merged**